drivers/volume/portworx/portworx.go

```diff
@@ -330,6 +312,67 @@ func (d *portworx) WaitStart(n node.Node) error {
 	return err
 }
 
+func (d *portworx) setDriver() error {
+	var err error
+	nodes := d.schedDriver.GetNodes()
+
+	var endpoint string
+	// Try portworx-service first
+	svc, err := k8sops.Instance().GetService(PXServiceName, PXNamespace)
+	if err == nil {
+		endpoint = svc.Spec.ClusterIP
+		if ok, err := d.testEndpoint(endpoint); ok && err == nil {
+			err = d.setClientsForEndpoint(endpoint)
+			return err
+		}
+	}
+
+	// Try direct address of cluster nodes
+	for _, n := range nodes {
+		if n.Type == node.TypeWorker {
+			for _, addr := range n.Addresses {
+				if ok, err := d.testEndpoint(addr); ok && err == nil {
```

Comment:
should we club the test and setClient method together as before as we are creating a client anyway?
this way if there is an error we proceed to the next address and not stop checking

Reply:
Fixed in incremental